Sources/ZcashLightClientKit/Rust/ZcashRustBackend.swift

```
@@ -100,6 +103,35 @@ class ZcashRustBackend: ZcashRustBackendWelding {
         return true
     }
 
+    static func isValidExtendedSpendingKey(_ key: String, networkType: NetworkType) throws -> Bool {
+        guard !key.containsCStringNullBytesBeforeStringEnding() else {
+            return false
+        }
+
+        guard zcashlc_is_valid_sapling_extended_spending_key(key, networkType.networkId) else {
```

If we want to have network-incompatibility identified as a specific failure case that the SDK can recognize, we should change the implementation of this FFI call to return an integer value (0 for success, positive nonzero for "known" errors that the caller should handle, -1 for unrecoverable errors) instead. However, if we wanted to do that we'd have to change how we do parsing in librustzcash, because right now we discard the reason for the parse failure. @str4d do you have an opinion here? https://github.com/zcash/librustzcash/blob/main/zcash_client_backend/src/address.rs#L203 swallows the specific error type; should we return `Result` from these methods instead?